app/tests/spec/views/mixins/settings-panel-mixin.js

```diff
@@ -76,6 +77,57 @@ define(function (require, exports, module) {
         assert.isFalse(view.isPanelOpen());
       });
 
+      it('openPanel focuses the first input field if present', function () {
+        // create and append an input field
+        var $dummyInput = $('<input type="text" name="dummyholder">');
+        view.$('.settings-unit').append($dummyInput);
+        view.openPanel();
+
+        // input field should be present, we just appended it
+        var $input = view.$('.open input:first-of-type');
+        assert.isTrue($input.length === 1, 'Input field not present');
+        // input field should have been focused
+        assert.equal($input[0], document.activeElement, 'Input does not have focus');
+      });
+
+      it('openPanel focuses the first button field when input is not present', function () {
+        // for button to be focused, input field should not be present
+        var $input = $('.open input:first-of-type');
+        assert.isTrue($input.length === 0, 'Input element is present');
+
+        var $dummyButton = $('<button type="submit" class="submit primary">Subscribe</button>');
+        view.$('.settings-unit').append($dummyButton);
+        view.openPanel();
+        // button should be present, we just appended it
+        var $button = $('.open button.primary');
+        assert.isTrue($button.length > 0, 'Button not found');
+        // button should have been focused
+        assert.equal($button[0], document.activeElement, 'Button does not have focus');
+      });
+
+      it('_hidePanelOnEscape calls hidePanel when escape key is pressed', function () {
+        sinon.stub(view, 'hidePanel', function () {});
+        // keycode for escape key
+        view._hidePanelOnEscape(KeyCodes.ESCAPE);
```

Comment:
Looks like the code coverage went down a [little](https://coveralls.io/builds/6682555/source?filename=app%2Fscripts%2Fviews%2Fmixins%2Fsettings-panel-mixin.js). You could use `view.onKeyUp(KeyCodes.ESCAPE)` to go through the extra path.